src/Database/Connection.php

```diff
@@ -919,7 +920,7 @@ public function getLogger(): LoggerInterface
             return $this->_logger;
         }
 
-        if (!class_exists(QueryLogger::class)) {
+        if (!class_exists(BaseLog::class)) {
```

Comment:
This hidden dependency isn't very maintainable. I wonder if we can attach it to something.

Reply:
Yeah, it's not overly nice, but at the moment I don't have any overly good ideas. There's more of those checks btw, for caching and for `CSPBuilder`.

Reply:
Yea, but this checks for a class that isn't being directly instantiated. That's all I'm saying. We probably should merge and fix this as-is though.